sips/network_topology.md

```diff
@@ -0,0 +1,202 @@
+
+| Committee Size ($k$) | Probability ($P(t(C_1) = t(C_2))$) |
+|:---------------------|:-----------------------------------|
+| 4                    | 60.31%                             |
+| 7                    | 75.20%                             |
+| 10                   | 81.96%                             |
+| 13                   | 85.82%                             |
+
+> [!NOTE]
+> Alternative [Locality Sensitive Hashing (LSH)](https://en.wikipedia.org/wiki/Locality-sensitive_hashing) functions were tested but proved unsuitable for our context:
+> - **SimHash (Charikar's Hash)**:
+> While SimHash theoretically approximates Cosine Similarity,
+> which offers a higher collision probability than MinHash (e.g. ~77% vs. 60% for 4-element sets),
+> it failed in practice because it produces a "fingerprint" for the entire set rather than selecting a common element in a small discrete set.
+> Even though similar committees produce similar fingerprints, a single bit difference, caused by changing one operator, results in a completely different integer value
+> (and thus resulting topic).
+> - **Consensus MinHash (Voting) and Multi-Hash Minimum**: We attempted to boost stability by (i) computing k independent MinHashes and taking a majority vote,
+> and (ii) by taking the global minimum of the $k$ hashes. 
+> However, this offered no improvement and only incurred more computation overhead.
+
+## Spec Change
+For each committee, we assign its topic in the following way:
+
+```javascript
+CONSTANTS:
+NUM_TOPICS = 128
+MAX_UINT256 = (2^256) - 1
+
+FUNCTION AssignCommitteeToTopic(committee: list[uint64]):
+	min_int = MAX_UINT256
+	
+	FOR each operator_id IN committee:
+		# 1. Serialize the uint64 Operator ID to 8 bytes (Little-Endian)
+		input_bytes = serialize_uint64_little_endian(operator_id)
+		
+		# 2. Compute the SHA-256 hash, resulting in 32 bytes
+		hash_bytes = SHA256(input_bytes)
+		
+		# 3. Convert to uint256 (Big-Endian)
+		current_int = uint256_from_bytes_big_endian(hash_bytes)
+		
+		# 4. Get minimum
+		IF current_int < min_int:
+			min_int = current_int
+	
+	# 5. Modulo for topic ID
+	RETURN min_int MOD NUM_TOPICS
+```
+
+## Fork Transition
+
+To safely transition through the fork, we define the transition policy below. Let:
+- $old$: denotes the set of topics the operator should be subscribed to before the fork (according to its committees),
+- $new$: denotes the set of topics the operator should subscribe to after the fork.
+
+### Before the Fork
+
+The operator is subscribed to the $old$ topics and publishes to them.
+
+`PRIOR_WINDOW` epochs before the fork, it subscribes to all $new + old$.
+During this window, it does **not reject** any message with a valid signature in the new topics ($new \setminus old$).
+
+We set `PRIOR_WINDOW = 2` epochs to allow
+enough time for the operator to set up the new topics before the fork,
+and avoid a long overloading period (due to the extra messages being processed in the new topic).
```

Comment:
Why the extra messages? Honest nodes shouldn't publish to the new topics. But, true, there might be some overlap between the new and old topics, and nodes will see traffic in subnets that are their new ones. Those messages should be ignored as in the wrong topic.

Reply:
Yes, I'm referring to the same context. It might be worth elaborating a bit more on the spec to make it clearer. 

Should the new nodes propagate these messages? Shouldn't the mesh continue working fine as it did before the new nodes joined?

Reply:
we should also mention warming up the mesh, it is an important consideration.

Reply:
we should also mention warming up the mesh, it is an important consideration.

Reply:
@MatheusFranco99 You're right, we need to propagate if by warming up we mean start behaving like a normal peer in that mesh. But we should make the distinction that we'll accept but not process messages during this period, as they aren't relevant. I also believe we need to make sure that doubling the node load (regarding validation and propagation) won't impact its duties.